apollo-federation/src/supergraph/mod.rs

```diff
@@ -747,6 +766,45 @@ fn extract_object_type_content(
             )?;
         }
 
+        if let Some((context_spec_def, name_in_supergraph)) = context {
+            for directive in type_.directives.get_all(name_in_supergraph.as_str()) {
+                if let Some(context_name) =
+                    FederationSpecDefinition::context_directive_arguments(directive).ok()
+                {
+                    // todo should I return an error if there isn't a value?
+                    let arr: std::str::Split<'_, &str> = context_name.name.split("__");
+                    if let Some(subgraph_name) = arr.next() {
+                        let subgraph_name = Name::new(subgraph_name)?;
+                        let subgraph = get_subgraph(
+                            subgraphs,
+                            graph_enum_value_name_to_subgraph_name,
+                            &subgraph_name,
+                        )?;
+                        let federation_spec_definition = federation_spec_definitions
+                            .get(&subgraph_name)
+                            .ok_or_else(|| SingleFederationError::InvalidFederationSupergraph {
+                                message: "Subgraph unexpectedly does not use federation spec"
+                                    .to_owned(),
+                            })?;
+                        if let Some(context_in_subgraph) = arr.last() {
+                            pos.insert_directive(
+                                &mut subgraph.schema,
+                                (federation_spec_definition.context_directive(
+                                    &subgraph.schema,
+                                    context_in_subgraph.to_string(),
+                                )?)
+                                .into(),
+                            );
+                        } else {
+                            todo!();
+                        }
+                    } else {
+                        todo!();
+                    }
```

Comment:
Could you give an example of what it would look like to reach these else branches? It feels like these should be errors (like you mention in your comment), but I'm not sure what the error would be.